operator-common/src/test/java/io/strimzi/operator/common/ReconnectingWatcherMockTest.java

```diff
@@ -0,0 +1,246 @@
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@EnableKubernetesMockClient(crud = true)
+public class ReconnectingWatcherMockTest {
+    private static final Logger LOGGER = LogManager.getLogger(ReconnectingWatcherMockTest.class);
+
+    private static final String NAMESPACE = "my-namespace";
+    private static final String NAMESPACE2 = "my-namespace2";
+    private static final String CLUSTER_NAME = "my-cluster";
+    private static final String CLUSTER_NAME2 = "my-cluster2";
+
+    private Vertx vertx;
+    // Injected by Fabric8 Mock Kubernetes Server
+    @SuppressWarnings("unused")
+    private KubernetesClient client;
+    private MockKube2 mockKube;
+
+    @BeforeEach
+    public void beforeEach() {
+        vertx = Vertx.vertx();
+
+        // Configure the Kubernetes Mock
+        mockKube = new MockKube2.MockKube2Builder(client)
+                .withKafkaCrd()
+                .build();
+        mockKube.start();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        mockKube.stop();
+        vertx.close();
+    }
+
+    @Test
+    public void testWatch() throws InterruptedException {
+        CrdOperator<KubernetesClient, Kafka, KafkaList> kafkaOps = new CrdOperator(vertx, client, Kafka.class, KafkaList.class, Kafka.RESOURCE_KIND);
+
+        CountDownLatch addedLatch = new CountDownLatch(1);
+        CountDownLatch modifiedLatch = new CountDownLatch(1);
+        CountDownLatch deletedLatch = new CountDownLatch(1);
+        AtomicInteger eventCounter = new AtomicInteger(0);
+
+        ReconnectingWatcher<Kafka> watcher = new ReconnectingWatcher<>(kafkaOps, Kafka.RESOURCE_KIND, NAMESPACE, null, (a, r) -> {
+            LOGGER.info("Received event {} about resource {}", a, r.getMetadata().getName());
+
+            switch (a)  {
+                case ADDED -> {
+                    addedLatch.countDown();
+                    eventCounter.incrementAndGet();
+                }
+                case MODIFIED -> {
+                    modifiedLatch.countDown();
+                    eventCounter.incrementAndGet();
+                }
+                case DELETED -> {
+                    deletedLatch.countDown();
+                    eventCounter.incrementAndGet();
+                }
+            }
+        });
+
+        Kafka kafka = new KafkaBuilder()
+                .withNewMetadata()
+                    .withName(CLUSTER_NAME)
+                .endMetadata()
+                .withNewSpec()
+                    .withNewKafka()
+                        .withReplicas(1)
+                    .endKafka()
+                .endSpec()
+                .build();
+
+        Crds.kafkaOperation(client).inNamespace(NAMESPACE).resource(kafka).create();
+        Crds.kafkaOperation(client).inNamespace(NAMESPACE2).resource(kafka).create();
+        addedLatch.await();
```

Comment:
We should set a timeout for all await calls, in order to avoid a stuck test in case someone breaks the ReconnectingWatcher and the handler is never called.

Reply:
Good point, fixed.